apps/ui/src/modules/profile/components/profileCards.tsx

```diff
@@ -49,33 +52,64 @@ export const ProfileCards = ({
     </Suspense>
   );
 
-  const LoadedData = () => (
-    <Suspense>
-      <Stack
-        display="flex"
-        h="fit-content"
-        spacing={6}
-        direction={['column', 'column', 'column', 'row']}
-        w="full"
-      >
-        <Flex w="full" h="full" flexDirection="column" gap={[4, 4, 4, 6]}>
-          <ProfileCard domainName={domainParam} domain={domain ?? ''} />
+  const LoadedData = () => {
+    const { domain: domainName } = useParams({ strict: false });
+    const { wallet } = useWallet();
+    const [metadata, setMetadata] = useState<Metadata[] | undefined>();
 
-          <Stack
-            w="full"
-            h="full"
-            direction={['column', 'column', 'column', 'row']}
-            gap={[4, 4, 4, 6]}
-          >
-            <OwnershipCard owner={owner ?? ''} />
+    // biome-ignore lint/correctness/useExhaustiveDependencies: <explanation>
+    useEffect(() => {
+      if (!wallet) return;
 
-            <AddressesCard domain={domain ?? ''} />
-          </Stack>
-        </Flex>
-        <AccountsCard />
-      </Stack>
-    </Suspense>
-  );
+      try {
+        const userMetadata = UserMetadataContract.initialize(
+          wallet,
+          domainName,
+        );
+
+        userMetadata.getAll().then((metadata) => {
+          console.log(metadata);
+          setMetadata(metadata);
+        });
+      } catch (error) {
+        console.log(error);
+      }
```

Comment:
how are you managing the loading and error states? 
It’s better to use react-query for better control over this

Reply:
```ts
  const { data: metadata, isLoading, isError, error } = useQuery({
    queryKey: ['QUERY_NAME'],
    queryFn: async () => {
      const userMetadata = UserMetadataContract.initialize(
        wallet,
        domainName,
      );

      return userMetadata.getAll()
    },
  });
```

Reply:
Good point, i'll be changing for react-query call.